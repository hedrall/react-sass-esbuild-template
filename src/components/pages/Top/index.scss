@use '../../../assets/variables' as v;

.Top {
  padding: 32px;

  .Title {
    margin-bottom: 16px;
  }

  .Counter {
    padding: 16px;
    border: #929292 solid 1px;
    border-radius: 7px;
    width: 200px;
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 8px;

    > div {
      text-align: center;
    }

    button {
      width: 120px;
      background: v.$button-color;
      color: white;
      border: none;
      border-radius: 3px;
      padding: 2px 8px;
    }
  }
}