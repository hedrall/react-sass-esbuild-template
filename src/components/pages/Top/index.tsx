import React from "react";
import { useRecoilState } from "recoil";
import { sampleState as _sampleState } from "@frontend/store/atoms";

export type TopProps = {};

export const Top: React.FC<TopProps> = (props) => {
  const [sampleState, setSampleState] = useRecoilState(_sampleState);

  const onClickHandler = () =>
    setSampleState((pre) => ({ count: pre.count + 1 }));

  return (
    <div className="Top">
      <div className="Title">Top Page</div>

      <div className="Counter">
        <div>{sampleState.count}</div>
        <button onClick={onClickHandler}>count up</button>
      </div>
    </div>
  );
};
