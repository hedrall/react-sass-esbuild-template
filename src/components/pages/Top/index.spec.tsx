import React from 'react';
import { render, screen } from '@testing-library/react/pure';
import { Top, TopProps } from '@frontend/components/pages/Top/index';

describe('Top', () => {
  
  let container: HTMLElement;
  const props: TopProps = {

  }
  
  beforeAll(() => {
    const tree = render(<Top {...props} />);
    container = tree.container;
  });

  test('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
