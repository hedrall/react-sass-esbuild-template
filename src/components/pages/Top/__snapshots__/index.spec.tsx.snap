// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top snapshot 1`] = `
<div>
  <div
    class="Top"
  >
    <div
      class="Title"
    >
      Top Page
    </div>
    <div
      class="Counter"
    >
      <div>
        0
      </div>
      <button>
        count up
      </button>
    </div>
  </div>
</div>
`;
